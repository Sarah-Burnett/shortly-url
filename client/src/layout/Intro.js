import React from 'react';
import styled from 'styled-components';
import { breakpoint, PrimaryButton, white } from '../Styles';

const IntroWrapper = styled.section`
    height: auto;
    width: 100vw;
    background: ${white};
    padding-bottom: 15vh;
    @media (min-width: ${breakpoint}) {
        padding-left: 8vw;
        height: 100vh;
        display: flex;
        flex-direction: row-reverse;
        align-items: center;
    }
`
const IntroImg = styled.div`
    width: 100vw;
    height: auto;
    display: flex;
    align-items: center;
    justify-content: center;
    @media (min-width: ${breakpoint}) {
        width: 50vw;
    }
    img {
        width: 100%;
    }
`
const IntroText = styled.div`
    height: auto;
    width: 100vw;
    display: flex;
    flex-wrap: wrap;
    flex-direction: column;
    align-items: center;
    justify-content: space-evenly;
    padding-bottom: 5vh;
    text-align: center;
    @media (min-width: ${breakpoint}) {
        height: 60vh;
        width: 50vw;
        align-items: flex-start;
        text-align: left;
    }
    p {
        margin-top: 5vh;
        font-size: larger;
    }
    button {
        margin-top: 5vh;
        padding: 3vh 3vw;
    }
`


export const Intro = () => {
    return (
        <IntroWrapper>
            <IntroImg>
                <img src="./illustration-working.svg" alt="woman working at desk"/>
            </IntroImg>
           <IntroText>
                <h1>More than just shorter links</h1>
                <p>Build your brandâ€™s recognition and get detailed insights on how your links are performing.</p>
                <PrimaryButton>Get Started</PrimaryButton>
            </IntroText>
        </IntroWrapper>
    )
}
