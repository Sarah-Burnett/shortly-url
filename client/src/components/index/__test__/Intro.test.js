import React from "react";
import ReactDOM from "react-dom";
import Intro from "../Intro";
import { render, cleanup } from "@testing-library/react";
import renderer from "react-test-renderer";

afterEach(cleanup);

it("renders intro without crashing", () => {
	const div = document.createElement("div");
	ReactDOM.render(<Intro />, div);
});

it("renders intro correctly", () => {
	const { getByRole } = render(<Intro />);
	expect(getByRole("heading").textContent).toBe("More than just shorter links");
	expect(getByRole("button").textContent).toBe("Get Started");
});

it("intro matches snapshot", () => {
	const tree = renderer.create(<Intro />).toJSON();
	expect(tree).toMatchSnapshot();
});
