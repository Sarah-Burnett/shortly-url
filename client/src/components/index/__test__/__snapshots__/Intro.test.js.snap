// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`intro matches snapshot 1`] = `
<section
  className="sc-fzplWN bqyiPU"
  data-testid="home"
>
  <div
    className="sc-fznyAO vcaiD"
  >
    <img
      alt="woman working at desk"
      src="illustration-working.svg"
    />
  </div>
  <div
    className="sc-fznKkj jsUFqS"
  >
    <h1>
      More than just shorter links
    </h1>
    <p>
      Build your brandâ€™s recognition and get detailed insights on how your links are performing.
    </p>
    <button
      className="sc-AxirZ dVJbcR"
    >
      Get Started
    </button>
  </div>
</section>
`;
